Basic methods v1.9

zbohm opened the pull request:
Minimální aktualizace současného kódu tak, aby fungoval na API 1.9. 
Modul kvůli jiné hešovací funkci již nefunguje s API 1.7.

Aktualizace zahrnuje komit z PR #10 opravující zavírání otevřených souborů. Dále zahrnuje PR #11 pro logování komunikace s API.

 * U podpisů natavena hešovací funkce `SHA256` místo původní `SHA`.
 * Do funcke `payment_init` doplněny další paramerty pro API. Funkce ale zůstává kompatibilní s verzí 0.7.0.
 * Funkce `oneclick_init` a `oneclick_start` odstraněny. Tím je klient uzpůsoben jen k základním metodám. Jiné metody, jako je OneClick, se budou se řešit později.
 * Do `HTTP_STATUSES` doplněny HTTP návratové kódy 401 a 503.

-----
    
Minimal update of current code to work on API 1.9.
The module no longer works with API 1.7 due to a different hash function.

The update includes a commit from PR #10 fixing the closing of open files. It also includes PR #11 for logging communication with the API.
    
 * Use `SHA256` hashing function for signatures instead of the unsafe `SHA`.
 * Additional parameters for the API have been added to the `payment_init` function. But the function remains compatible with version 0.7.0.
 * Functions `oneclick_init` and `oneclick_start` removed. This means that the client implements only Basic Payment. Other payments methods, such as OneClick, will be added later.
 * Added HTTP return codes 401 and 503 to `HTTP_STATUSES`.

zbohm committed:
Logging requests and responses.

JiriKr committed:
fixed #10

zbohm committed:
More about logging in README.

zbohm committed:
Upgrade the current code to version 1.9.

Minimal update of current code to work on API 1.9.
The module no longer works with API 1.7 due to a different hash function.

 * Use `SHA256` hashing function for signatures instead of the unsafe `SHA`.
 * Additional parameters for the API have been added to the `payment_init` function. But the function remains compatible with version 0.7.0.
 * Functions `oneclick_init` and `oneclick_start` removed. This means that the client implements only Basic Payment. Other payments methods, such as OneClick, will be added later.
 * Added HTTP return codes 401 and 503 to `HTTP_STATUSES`.

zbohm committed:
Upgrade function button to button_init for API v1.9.

zbohm commented:
Funkce `button` aktualizována na API 1.9 a přejmenována na `button_init`. Tím je kromě `Basic Payments` implementována i část `Methods for ČSOB Payment Button`.

tdivis commented on pycsob/conf.py:
```diff
@@ -69,11 +69,13 @@
-    503: 'Service Unavailable'
+    500: 'Internal Server Error',
+    503: 'Service Unavailable',
 }
```
Is this `HTTP_STATUSES` variable used anywhere?

zbohm replied:
No, it is not in use anywhere yet.

tdivis commented on pycsob/client.py:
```diff
@@ -71,9 +68,10 @@ def payment_init(self, order_no, total_amount, return_url, description, cart=Non
         :param return_url: URL to be returned to from payment gateway
         :param cart: items in cart, currently min one item, max two as mentioned in CSOB spec
-        :param description: order description
+        :param description: product name
```
I would add that it's only used for the cart item name if cart is emtpy.

tdivis replied:
There should be changelog entry with incompatible and other changes/migration guide.

zbohm replied:
Docstring and Changelog  have been updated.

tdivis commented:
I'm missing the update of RETURN_CODES as there was quite a few changes in 1.8 and 1.9 judging from [changleog](https://github.com/csob/paymentgateway/blob/master/CHANGELOG.md)

zbohm committed:
Update CHANGELOG.

oto-stefan committed:
Fix typo in class name

'PyscobSession' -> 'PycsobSession' (as CSOB is the bank name)

oto-stefan committed:
Add support for custom payments

'payment/init' request: 'payOperation' with value 'customPayment'
  and 'customExpiry' items already supported
'payment/init' response: add 'customerCode' item

oto-stefan committed:
Update README with basic custom payment instructions

zbohm committed:
Adjust long_description_content_type in setup.

zbohm committed:
Add bumpversion.cfg.

zbohm committed:
Use user agent with current client version in tests.

zbohm committed:
Bump version: 1.0.0 → 1.0.1

oto-stefan committed:
Sign properly complex data (#2)

* Sign properly complex data

Compose message from input data for signing accounting for nested
sequences and mappings.

* fixup - Correction from code review

* fixup - Fix typo

* fixup - Fix typo in docstring

* fixup - Code review (round 2)

* fixup - Add docs comment to enums

* fixup - Unnecessary trimming

* fixup - strip after trimming

* Correction from review - round 3

* fixup - Corrections from review (round 4)

* fixup - Change date/time related fields

* fixup - Custom formatting should have higher priority

* fixup - Add docstrings

* Improve type annotations

* Use all amounts always as integers

* fixup - README and type annotation

* fixup - Add enumerations

* Remove Python 3.8

zbohm committed:
Add into setup python_requires >=3.9.
